src/modularity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+    randompartitioning(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
+    
+Generates a random `n`-partitioning for a hypergraph `hg`. 
+"""
+function randompartitioning(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
+    x = rand(1:n, size(hg, 1))
+    [findall(==(i), x) for i in 1:n]
+end
+
+"""
+    modularity(hg::Hypergraph, partitioning::Vector{Vector{Int}})
+    
+Calculates the modularity of a hypergraph `hg` for a given `partitioning`.
+"""
+function modularity(hg::Hypergraph, partitioning::Vector{Vector{Int}})
+    @assert sum(length.(partitioning)) == size(hg, 1)
+    @assert sort!(union(partitioning...)) == axes(hg, 1)
+    hes = [count(==(true), hg[:, i]) for i in axes(hg, 2)]
+    Ed = Dict{Int,Int}()
+    for i in 2:maximum(hes)
+        Ed[i] = count(==(i), hes)
+    end
+    deg(hg, i) = count(==(true), hg[i, :])
+    volV = 2*size(hg,2)
+    volP = [sum(deg(hg, i) for i in p) for p in partitioning]
+    eP = [count(i->issubset(findall(==(true), hg[:,i]), p)*any(==(true),hg[:,i]), axes(hg,2)) for p in partitioning]
+    (sum(eP) - sum(Ed[d]*sum((v/volV)^d for v in volP) for d in keys(Ed))) / size(hg, 2)
+end
+
+"""
+    modularity(hg::Hypergraph)
+    
+Calculates the modularity of a hypergraph `hg`.
+"""
+modularity(hg::Hypergraph) = modularity(hg, [collect(1:size(hg, 1))])
```

Comment:
why do you call it `modularity`?

Reply:
it calculates modularity assuming partitioning into a single partition containig all nodes.

Reply:
I know. But is it really needed to have this method? Normally when I hear "graph modularity" I think of a maximum possible value of modularity considering all possible partitions.

Reply:
we have `findmodularity` function for that. Let me know if you want to have different naming.

Reply:
Please ask Pawel what he thinks. We can have it as you propose.

Reply:
I assume that the current approach is something Pawel thinks is OK?

Reply:
I discussed it with him

Reply:
OK, so can we put somewhere in the documentation the target API specification that you have agreed on so that we have a reference to check against? In particular I understand that we will have `modularity` and `findmodularity` as key defined functions - right? How are we going to handle non-strict modularity, given Pawel's question about it?